src/Flac/FlacConverter.php

```diff
@@ -51,119 +56,21 @@ public function setSupportedOutputFormats(array $supportedOutputFormats): void
-
-        if ($fileDetails['playtime_seconds'] < $clipLength) {
-            // No trim or fades are necessary if the play-time
-            // is less than the specified clip length.
-
-            $performTrim = false;
-            $canFadeIn = false;
-            $canFadeOut = false;
-        }
-
-        // This block prevents problems from a track preview start
-        // time that is too far into the track to allow for a preview clip of
-        // the specified length.
-
-        if (($fileDetails['playtime_seconds'] - $trimStartTime) < $clipLength) {
-            // We'll force the track preview to start exactly $clipLength seconds
-            // before the end of the track, thus forcing a $clipLength-second preview
-            // clip length.
-
-            $trimStartTime = $fileDetails['playtime_seconds'] - $clipLength;
-        }
-
-        // Convert the clip length from seconds to hh:mm:ss format.
-
-        $clipLength = Utility::sec2hms($clipLength, false, false);
-        */
-
-        $cmd = [];
+        $this->flacEffect->command->input($inputFile);
 
-        // Attempt to perform the transcoding operation.
+        $this->flacEffect->command->output($outputFile);
 
-        // TODO Deal with this.
-
-        #if ($this->singleThreaded === true) {
-            $this->effect->command->singleThreaded();
-        #}
-
-        // If setlocale(LC_CTYPE, "en_US.UTF-8") is not called here, any UTF-8 character will equate to an empty string.
-
-        setlocale(LC_CTYPE, 'en_US.UTF-8');
-
-        $this->effect->command->verbosity(4);
-
-        $this->effect->command->input($inputFile);
-
-        $this->effect->command->channels(2);
-
-        $this->effect->command->output($outputFile);
-
-        // TODO Deal with this.
-
-        /*
-        if ($performTrim !== false) {
-            $cmd[] = ' trim ' . escapeshellarg($trimStartTime) . ' ' . escapeshellarg($clipLength);
-        }
-
-        if ($canFadeIn !== false || $canFadeOut !== false) {
-            $cmd[] = ' fade q ';
-
-            if ($canFadeIn === false) {
-                // Setting a fade-in length of zero in SoX is the
-                // same as having no fade-in at all.
-
-                $fadeInLength = '0';
-            }
-
-            $cmd[] = escapeshellarg($fadeInLength);
-
-            if ($canFadeOut !== false) {
-                $cmd[] = ' ' . escapeshellarg($clipLength) . ' ' . escapeshellarg($fadeOutLength);
-            }
-        }
-        */
-
-        // If "['LC_ALL' => 'en_US.utf8']" is not passed here, any UTF-8 character will appear as a "#" symbol.
+        // If "['LC_ALL' => 'en_US.utf8']" is not passed here, any UTF-8
+        // character will appear as a "#" symbol.
 
         $env = ['LC_ALL' => 'en_US.utf8'];
 
-        $this->process->setCommand($this->effect->getCommand()->compose());
+        $this->process->setCommand($this->flacEffect->getCommand()->compose());
 
         $this->process->setTimeout(600);
```

Comment:
Why this ?

Reply:
This is precisely what I wanted to discuss.

If you look at the `toAlac()` method body, you will see `$this->alacEffect->...`.

Basically, we have the need to use two different `Effect` classes within the `FlacConverter` class: one for FLAC files and one for ALAC files.

Prior to this commit, we had only needed one (for FLAC files), so `$this->effect->...` was sufficient.

I'm struggling to visualize a better solution to this need...

We discussed this while brainstorming in a side-channel and made some notes in the Effect class constructors, e.g.: https://github.com/indiehd/audio-manipulator/pull/3/files#diff-1c1ba449d97f8c333ead259f08b7e988R17

I feel like I'm missing something obvious here. Throw me a bone? :D

Reply:
Resolving (pardon the pun!) this issue seems better-suited for a separate PR. If you're agreeable to it, let's resolve this conversation and address this in a subsequent PR.